packages/flutter/lib/src/widgets/animated_scroll_view.dart

```diff
@@ -83,51 +196,192 @@ class _ActiveItem implements Comparable<_ActiveItem> {
+  ///
+  /// See also:
+  ///
+  ///  * [of], a similar function that will throw if no [AnimatedGrid] ancestor
+  ///    is found.
+  static AnimatedGridState? maybeOf(BuildContext context) {
+    assert(context != null);
+    return context.findAncestorStateOfType<AnimatedGridState>();
+  }
+
+  @override
+  AnimatedGridState createState() => AnimatedGridState();
+}
+
+/// The state for a scrolling container that animates items when they are
+/// inserted or removed.
+///
+/// When an item is inserted with [insertItem] an animation begins running. The
+/// animation is passed to [AnimatedGrid.itemBuilder] whenever the item's widget
+/// is needed.
+///
+/// When an item is removed with [removeItem] its animation is reversed.
+/// The removed item's animation is passed to the [removeItem] builder
+/// parameter.
+///
+/// An app that needs to insert or remove items in response to an event
+/// can refer to the [AnimatedGrid]'s state with a global key:
+///
+/// ```dart
+/// // (e.g. in a stateful widget)
+/// GlobalKey<AnimatedGridState> gridKey = GlobalKey<AnimatedGridState>();
+///
+/// // ...
+///
+/// @override
+/// Widget build(BuildContext context) {
+///   return AnimatedGrid(
+///     key: gridKey,
+///     itemBuilder: (BuildContext context, int index, Animation<double> animation) {
+///       return const Placeholder();
+///     },
+///     gridDelegate: const SliverGridDelegateWithMaxCrossAxisExtent(maxCrossAxisExtent: 100.0),
+///   );
+/// }
+///
+/// // ...
+///
+/// void _updateGrid() {
+///   // adds "123" to the AnimatedGrid
+///   gridKey.currentState!.insertItem(123);
+/// }
+/// ```
+///
+/// [AnimatedGrid] item input handlers can also refer to their [AnimatedGridState]
+/// with the static [AnimatedGrid.of] method.
+class AnimatedGridState extends _AnimatedScrollViewState<AnimatedGrid> {
+
+  @override
+  Widget build(BuildContext context) {
+    return _wrap(
+      SliverAnimatedGrid(
+        key: _sliverAnimatedMultiBoxKey,
+        gridDelegate: widget.gridDelegate,
+        itemBuilder: widget.itemBuilder,
+        initialItemCount: widget.initialItemCount,
+      ),
+    );
+  }
+}
+
+abstract class _AnimatedScrollView extends StatefulWidget {
+  /// Creates a scrolling container that animates items when they are inserted
+  /// or removed.
+  const _AnimatedScrollView({
+    super.key,
+    required this.itemBuilder,
     this.initialItemCount = 0,
     this.scrollDirection = Axis.vertical,
     this.reverse = false,
     this.controller,
     this.primary,
     this.physics,
+    this.shrinkWrap = false,
     this.padding,
     this.clipBehavior = Clip.hardEdge,
-  })  : assert(itemBuilder != null),
+  }) : assert(itemBuilder != null),
         assert(initialItemCount != null && initialItemCount >= 0);
 
-  /// Called, as needed, to build grid item widgets.
+  /// {@template flutter.widgets.AnimatedScrollView.itemBuilder}
+  /// Called, as needed, to build list item widgets.
```

Comment:
Should this say "list item"? Maybe "container's" or "view's"?
(And several other places below...)